docs/web/usecases-internationalization.mdx

```diff
@@ -0,0 +1,280 @@
+<TabItem value="layout">
+
+```xml title="app/layout.xml"
+<?xml version="1.0" encoding="UTF-8"?>
+<layout xmlns="https://geocortex.com/layout/v1" xmlns:custom="your.custom.namespace">
+    <map>
+        <custom:translatable-text slot="top-center" width="15" height="9" padding="0.5"/>
+    </map>
+</layout>
+```
+
+</TabItem>
+<TabItem value="config">
+
+```json title="src/app.json"
+{
+    "items": []
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Creating the Language Resources
+
+Next, we are going to create the language resources for english and german. Language resources are plain json files with language keys and values.
+
+<Tabs
+  defaultValue="english"
+  values={[
+    { label: "English", value: "english" },
+    { label: "German", value: "german" },
+  ]}>
+<TabItem value="english">
+
+```json title="src/en.json"
+{
+    "language-translatable-text-content": "I will be translated."
+}
+```
+
+</TabItem>
+<TabItem value="german">
+
+```json title="src/de.json"
+{
+    "language-translatable-text-content": "Dieser Text wird übersetzt."
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Register the Language Resources
+
+Next we need to register the language resources with Geocortex Web. Geocortex Web uses two letter language codes to identify locales. See the [ISO 639-1 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) for a full list of possible locales.
+
+```ts title="src/index.ts"
+import { ComponentType } from "react";
+import { LibraryRegistry } from "@geocortex/web/config";
+import TranslatableText, {
+    TranslatableTextModel,
+} from "./components/TranslatableText";
+// highlight-start
+import enJson from "./en.json";
+import deJson from "./de.json";
+// highlight-end
+
+export default function (registry: LibraryRegistry) {
+    registry.registerComponent({
+        name: "translatable-text",
+        namespace: "your.custom.namespace",
+        getComponentType: () => TranslatableText as ComponentType,
+        itemType: "translatable-text",
+        title: "Translatable Text",
+    });
+    registry.registerModel({
+        getModelType: () => TranslatableTextModel,
+        itemType: "translatable-text",
+    });
+    // highlight-start
+    registry.registerLanguageResources({
+        locale: "en",
+        getValues: () => enJson,
+    });
+    registry.registerLanguageResources({
+        locale: "de",
+        getValues: () => deJson,
+    });
+    // highlight-end
+}
+```
+
+### Use the Language Strings
+
+Now that they have been registered, we can use the strings in our boilerplate component from earlier. To translate a language key, you can use the [`useContext` react hook](https://reactjs.org/docs/hooks-reference.html#usecontext) with the [UIContext](sdk-ui-context.mdx) to access the `translate` function. Then you can pass the language key `language-translatable-text-content` you defined in the language resources files to the `translate` function and your text will be translated into whichever locale is most appropriate.
```

Comment:
Can we reverse the order of these two paragraphs? If you're doing things right, you should almost never need to call translate() manually.

Reply:
reversed!